testsuite/features/support/api_test.rb

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2011-2022 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### Uncomment this line for XML-RPC API
+# require 'xmlrpc/client'
+
+### Uncomment these lines for HTTP API
+require 'faraday'
+require 'faraday/net_http'
+Faraday.default_adapter = :net_http
+
+def retrieve_server_id(server)
+  sysrpc = APISystemTest.new(ENV['SERVER'])
+  sysrpc.login('admin', 'admin')
+  systems = sysrpc.list_systems
+  refute_nil(systems)
+  server_id = systems
+              .select { |s| s['name'] == server }
+              .map { |s| s['id'] }.first
+  refute_nil(server_id, "client #{server} is not yet registered?")
+  server_id
+end
+
+### Uncomment these lines for HTTP API
+# wrapper for HTTP library (currently Faraday)
+class HttpClient
+  def initialize(url)
+    @faraday = Faraday.new(url, { headers: {'Content-Type' => 'application/json'} })
+  end
+
+  def call(name, *params)
+    puts params.to_json
+    ret = @faraday.post(name) do |req|
+      req.body = params.to_json
+    end
+    puts ret.status
+    puts ret.headers
+    puts ret.body
+    ret
+  end
+end
+
+# base class from where others inherit
+class APIBaseTest
```

Comment:
I would keep this as an abtract class.
Then have two classes, for instance, XMLRPCAPITest and HTTPAPITest which will implement initiallze.
With that approach, when we start the concrete Cucumber feature we instantiate a variable that will have feature scope, and this variable will be used as APIBaseTest but instantated with XMLRPC or HTTP, and the rest will be transparent.

Reply:
I mean, there are multiple ways to play with these classes so we can use both connections instead of commenting one or the other :)

Reply:
Sorry, this test is not ready for proofreading yet.

Besides, APIBaseTest describes a test. Http_client describes a connection to the server. They are of different natures.
It is considered a helper, and is accessed via a member object (`@connection`).

Commenting out is only one of the options. We could also use a global variable, for example. But at some point at run time you have to decide which implementation to use.

Reply:
I think we would just leave it all the time in HTTP form now on, but make it possible to switch back to XML-RPC any time with a single centralized change. I don't see the point of running both XML-RPC and HTTP at same time.

Reply:
Discussed in slack